src/clj/kixi/hecuba/data/measurements/calculations.clj

```diff
@@ -40,6 +45,22 @@
        (keep :value)
        (calculation-fn)))
 
+(defn extract-period
+  "Extracts data for a specific period.
+  Period can be :morning :day :evening or :night."
+  [period xs]
+  (let [timestamp   (->> xs first :timestamp (tc/to-date-time))
+        [start end] (get periods period)
+        start-date  (t/date-time (t/year timestamp) (t/month timestamp) (t/day timestamp)
+                                 (:hour start) (:minutes start))
+        end-date    (t/date-time (t/year timestamp) (t/month timestamp) (t/day timestamp)
+                                 (:hour end) (:minutes end))]
+    (filter #(let [t (tc/to-date-time (:timestamp %))]
+               (and (or (t/before? t end-date)
+                        (= t end-date))
+                    (or (t/after? t start-date)
+                        (= t start-date)))) xs)))
```

Comment:
why not use t/within? 

http://clj-time.github.io/clj-time/doc/clj-time.core.html#var-within.3F